File: src/main/java/no/nav/bidrag/beregn/kostnadsberegnetbidrag/beregning/KostnadsberegnetBidragBeregningImpl.java
package no.nav.bidrag.beregn.kostnadsberegnetbidrag.beregning;

import java.math.BigDecimal;
import java.math.RoundingMode;
import no.nav.bidrag.beregn.kostnadsberegnetbidrag.bo.GrunnlagBeregning;
import no.nav.bidrag.beregn.kostnadsberegnetbidrag.bo.ResultatBeregning;

public class KostnadsberegnetBidragBeregningImpl implements KostnadsberegnetBidragBeregning {

  @Override
  public ResultatBeregning beregn(GrunnlagBeregning grunnlagBeregning) {

    var resultat = grunnlagBeregning.getUnderholdskostnad().getBelop()
        .multiply(grunnlagBeregning.getBPsAndelUnderholdskostnad().getAndelProsent())
        .divide(BigDecimal.valueOf(1), -1, RoundingMode.HALF_UP)
        .subtract(grunnlagBeregning.getSamvaersfradrag().getBelop());

    return new ResultatBeregning(resultat);
  }
}

File: src/main/kotlin/no/nav/bidrag/beregn/kostnadsberegnetbidrag/beregning/KostnadsberegnetBidragBeregningImpl.kt
package no.nav.bidrag.beregn.kostnadsberegnetbidrag.beregning

import no.nav.bidrag.beregn.kostnadsberegnetbidrag.bo.GrunnlagBeregning
import no.nav.bidrag.beregn.kostnadsberegnetbidrag.bo.ResultatBeregning
import java.math.BigDecimal
import java.math.RoundingMode

class KostnadsberegnetBidragBeregningImpl : KostnadsberegnetBidragBeregning {
    override fun beregn(grunnlag: GrunnlagBeregning): ResultatBeregning {
        val resultat = grunnlag.underholdskostnad.belop
            .multiply(grunnlag.bPsAndelUnderholdskostnad.andelProsent)
            .divide(BigDecimal.valueOf(1), -1, RoundingMode.HALF_UP)
            .subtract(grunnlag.samvaersfradrag.belop)

        return ResultatBeregning(resultat)
    }
}
